invalid new-expression of abstract class type 'esphome::m5cores3_display::M5CoreS3Display'

# m5cores3_display component 

The [m5cores3_display](https://github.com/m5stack/M5CoreS3-Esphome/tree/main/components/m5cores3_display) display component no longer works:

`invalid new-expression of abstract class type 'esphome::m5cores3_display::M5CoreS3Display'` in [m5cores3_display.h](https://github.com/m5stack/M5CoreS3-Esphome/blob/main/components/m5cores3_display/m5cores3_display.h#L7)

I think there has been a change in in [ESPHome source code](https://github.com/esphome/esphome/tree/dev/esphome/components/ili9xxx) in the name of some methods in the [ili9xxx component](https://esphome.io/components/display/ili9xxx.html). This would explain why it no longer works with the custom component but with the generic component.

ESPHome expects a method named get_height_internal() whereas in the component source code the name is different [get_height()](https://github.com/m5stack/M5CoreS3-Esphome/blob/main/components/m5cores3_display/m5cores3_display.h#L7).

**Building error**
```Compiling .pioenvs/m5cores3/libb12/M5GFX/M5GFX.cpp.o
Compiling .pioenvs/m5cores3/libb12/M5GFX/lgfx/Fonts/IPA/lgfx_font_japan.c.o
src/main.cpp: In function 'void setup()':
src/main.cpp:689:56: error: invalid new-expression of abstract class type 'esphome::m5cores3_display::M5CoreS3Display'
   m5cores3_lcd = new m5cores3_display::M5CoreS3Display();
                                                        ^
In file included from src/esphome.h:30,
                 from src/main.cpp:3:
src/esphome/components/m5cores3_display/m5cores3_display.h:7:7: note:   because the following virtual functions are pure within 'esphome::m5cores3_display::M5CoreS3Display':
 class M5CoreS3Display : public display::Display {
       ^~~~~~~~~~~~~~~
In file included from src/esphome.h:18,
                 from src/main.cpp:3:
src/esphome/components/display/display.h:643:15: note: 	'virtual int esphome::display::Display::get_height_internal()'
   virtual int get_height_internal() = 0;
               ^~~~~~~~~~~~~~~~~~~
src/esphome/components/display/display.h:644:15: note: 	'virtual int esphome::display::Display::get_width_internal()'
   virtual int get_width_internal() = 0;
               ^~~~~~~~~~~~~~~~~~
*** [.pioenvs/m5cores3/src/main.cpp.o] Error 1
Compiling .pioenvs/m5cores3/libb12/M5GFX/lgfx/Fonts/efont/lgfx_efont_cn.c.o
========================= [FAILED] Took 98.41 seconds =========================
```

**Full log**

[logs_cores3-voice-assistant_compile.txt](https://objects.githubusercontent.com/github-production-repository-file-5c1aeb/738759217/14816945?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240519%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240519T081011Z&X-Amz-Expires=300&X-Amz-Signature=28eaaf29f55c70b994f7181d8ed922f54a737e7c1fe2cce0beacd83c11c978f4&X-Amz-SignedHeaders=host&actor_id=152775900&key_id=0&repo_id=738759217&response-content-disposition=attachment%3Bfilename%3Dlogs_cores3-voice-assistant_compile.txt&response-content-type=text%2Fplain)

--- a/components/m5cores3_display/m5cores3_display.cpp
+++ b/components/m5cores3_display/m5cores3_display.cpp
@@ -39,8 +39,8 @@ void M5CoreS3Display::setup() {
 
 float M5CoreS3Display::get_setup_priority() const { return setup_priority::HARDWARE; }
 void M5CoreS3Display::dump_config() {}
-int M5CoreS3Display::get_width() { return M5.Display.width(); }
-int M5CoreS3Display::get_height() { return M5.Display.height(); }
+int M5CoreS3Display::get_width_internal() { return M5.Display.width(); }
+int M5CoreS3Display::get_height_internal() { return M5.Display.height(); }
 
 
 void M5CoreS3Display::draw_pixel_at(int x, int y, Color color) {

--- a/components/m5cores3_display/m5cores3_display.h
+++ b/components/m5cores3_display/m5cores3_display.h
@@ -9,8 +9,8 @@ class M5CoreS3Display : public display::Display {
   void setup() override;
   float get_setup_priority() const override;
   void dump_config() override;
-  int get_width() override;
-  int get_height() override;
+  int get_width_internal() override;
+  int get_height_internal() override;
 
   display::DisplayType get_display_type() override { return display::DisplayType::DISPLAY_TYPE_COLOR; }
   void fill(Color color) override;

